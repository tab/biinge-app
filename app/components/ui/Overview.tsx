import React from "react"
import { View } from "react-native"
import { useTranslation } from "react-i18next"

import { overviewStyles } from "styles"
import Typography from "./Typography"

type Props = {
  children: React.ReactNode
}

const OverviewComponent = ({ children }: Props) => {
  const { t } = useTranslation()
  return (
    <View style={overviewStyles.root}>
      <Typography variant="callout" style={overviewStyles.title}>
        {t("movie.content.overview")}
      </Typography>
      <Typography variant="body">{children}</Typography>
    </View>
  )
}

export default OverviewComponent
