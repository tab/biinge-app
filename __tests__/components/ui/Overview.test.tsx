import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import Overview from "components/ui/Overview"

describe("Overview", () => {
  const text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit."

  it("renders correctly", () => {
    const tree = render(<Overview>{text}</Overview>)
    expect(tree).toMatchSnapshot()
  })
})
