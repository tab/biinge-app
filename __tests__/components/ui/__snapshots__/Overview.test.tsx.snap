// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders correctly 1`] = `
<View
  style={
    {
      "padding": 0,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      [
        {
          "color": "#181818",
          "fontSize": 18,
          "fontWeight": "600",
          "lineHeight": 24,
        },
        {
          "color": "#666666",
          "marginBottom": 7,
        },
      ]
    }
  >
    Synopsis
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      [
        {
          "color": "#181818",
          "fontSize": 18,
          "fontWeight": "300",
          "lineHeight": 22,
        },
        {
          "color": "#181818",
        },
      ]
    }
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </Text>
</View>
`;
